parser/parser.go

```diff
@@ -506,21 +517,6 @@ func (p *parser) VisitUnary(ctx *gen.UnaryContext) interface{} {
 	return p.helper.newLiteralString(ctx, "<<error>>")
 }
 
-// Visit a parse tree produced by CELParser#MemberExpr.
-func (p *parser) VisitMemberExpr(ctx *gen.MemberExprContext) interface{} {
-	switch ctx.Member().(type) {
-	case *gen.PrimaryExprContext:
-		return p.VisitPrimaryExpr(ctx.Member().(*gen.PrimaryExprContext))
-	case *gen.SelectOrCallContext:
-		return p.VisitSelectOrCall(ctx.Member().(*gen.SelectOrCallContext))
-	case *gen.IndexContext:
-		return p.VisitIndex(ctx.Member().(*gen.IndexContext))
-	case *gen.CreateMessageContext:
-		return p.VisitCreateMessage(ctx.Member().(*gen.CreateMessageContext))
-	}
-	return p.reportError(ctx, "unsupported simple expression")
-}
```

Comment:
This is more of an observation but with this change, there is no longer a 1-to-1 correspondence with the visit implementations [with respect to the grammar](https://github.com/google/cel-go/blob/master/parser/gen/CEL.g4#L53), so the readability here does drop a bit.

Reply:
I think the docs on the `unnest` function will help a bit, but can understand how this degrades readability.